Modules/database_manager.py

```diff
@@ -0,0 +1,201 @@
+                condition(dict): a dict of conditions to filter the result.
+                    these will be connected via 'connector' and in the end must be true for the given row to be included in the result
+            Returns: list[tuple]
+                list of tuples, where each tuple represents the row, and each element of the tuple represents the value of the column
+            """
+            if not self._has_table(table_name): raise ValueError(f"Table {table_name} does not exist")
+
+            connector = connector.upper()
+            if connector not in ["AND", "OR"]: raise ValueError(f"Connector {connector} not supported")
+
+            condition_string: str = ""
+            for k, v in condition.items():
+                condition_string += f" {k} {v} {connector}"
+            if len(condition) > 0:
+                condition_string = condition_string[:-len(connector)]
+                condition_string = f" WHERE {condition_string}"
+
+            return self.connection.execute("SELECT * FROM '{tablename}' ?".format(tablename=table_name), (condition_string,)).fetchall() # table name cant be parameterized
+
+        def _update_row(self, table_name: str, connector: str, values: dict, condition: dict = {}):
+            """
+            Updates the rows of the given table with the given values filtering with the given conditions.
+            Args:
+                table_name (str): name of table to update
+                connector (str): 'AND' or 'OR'
+                values (dict): new value set. key is column name, value is new column value.
+                condition (dict): the set of conditions to be fulfilled to update the column
+
+            Returns: None
+            Raises: ValueError
+                If:
+                1.: The table does not exist
+                2.: No condition was given
+                3.: the Connector was not 'AND' or 'OR'
+            """
+            if not self._has_table(table_name): raise ValueError(f"Table {table_name} does not exist")
+
+            if len(condition) <= 0: raise ValueError("No conditions were given!")
+
+            connector = connector.upper()
+            if connector not in ["AND", "OR"]: raise ValueError("")
+
+            condition_string: str = ""
+            for k, v in condition.items():
+                condition_string += f" {k} {v} {connector}"
+            if len(condition_string) > 0:
+                condition_string = condition_string[:-len(connector)]
+                condition_string = f" WHERE {condition_string}"
+
+            value_string: str = ""
+            for k, v in values.items():
+                value_string += f" {k}={v},"
+
+            value_string = value_string[:-1]
+
+            self.connection.execute("UPDATE '{table_name}' SET (?) WHERE ?".format(table_name=table_name),  (value_string, condition_string))
+
+        def _delete_row(self, table_name: str, connector: str, condition: dict = {}):
```

Comment:
dictionary default parameter. this may produce unexpected behavior. this should be None and set to a empty dict within the method before use.